source/extensions/filters/network/geoip/geoip_filter.cc

```diff
@@ -53,9 +53,19 @@ Network::FilterStatus GeoipFilter::onNewConnection() {
   auto remote_address = read_callbacks_->connection().connectionInfoProvider().remoteAddress();
   ASSERT(driver_, "No driver is available to perform geolocation lookup.");
 
-  driver_->lookup(
-      Geolocation::LookupRequest{std::move(remote_address)},
-      [this](Geolocation::LookupResult&& result) { onLookupComplete(std::move(result)); });
+  // Capture weak_ptr to GeoipFilter so that filter can be safely accessed in the posted callback.
+  // This protects against the case when filter gets deleted before the callback is run
+  // (e.g., on LDS update).
+  GeoipFilterWeakPtr self = weak_from_this();
+  driver_->lookup(Geolocation::LookupRequest{std::move(remote_address)},
```

Comment:
nit: you may also want to add an integration test for lds update scenario (there is an example in http filter integration test).
otherwise lgtm, thanks!